### PR TITLE
TEXT(day_10): add find_next_neighbors tests

### PR DESCRIPTION
**Pull request by dariannax3**

### CODE REVIEW

File: day_10/task_1/tests/pipe_maze_tests.cpp [hunk 1]
```diff
@@ -5,5 +5,14 @@ TEST(PipeMazeToolsShould, FindTheStartingPoint)
...
+    ASSERT_EQ(find_starting_point(pipe_maze), MazePoint(1, 1));
+}
+
+TEST(PipeMazeToolsShould, FindeNextNeighbors)
+{
+    const PipeMaze pipe_maze = {"-L|F7", "7S-7|", "L|7||", "-L-J|", "L|-JF"};
+
+    const auto foundNeighbors = find_next_neighbors(pipe_maze);
+    ASSERT_EQ(foundNeighbors.first, (MazePoint(1, 2)));
+    ASSERT_EQ(foundNeighbors.second, (MazePoint(2, 1)));
```
Comment (Rosein) [inline day_10/task_1/tests/pipe_maze_tests.cpp]:
```suggestion
    ASSERT_EQ(foundNeighbors.first, MazePoint(1, 2));
    ASSERT_EQ(foundNeighbors.second, MazePoint(2, 1));
```

File: day_10/task_1/tests/pipe_maze_tests.cpp [hunk 2]
```diff
@@ -5,5 +5,14 @@ TEST(PipeMazeToolsShould, FindTheStartingPoint)
 {
     const PipeMaze pipe_maze = {"-L|F7", "7S-7|", "L|7||", "-L-J|", "L|-JF"};
 
-    ASSERT_EQ(find_starting_point(pipe_maze), (std::make_pair<int, int>(1, 1)));
+    ASSERT_EQ(find_starting_point(pipe_maze), MazePoint(1, 1));
+}
+
+TEST(PipeMazeToolsShould, FindeNextNeighbors)
```
Comment (Rosein) [inline day_10/task_1/tests/pipe_maze_tests.cpp]:
```suggestion
TEST(PipeMazeToolsShould, FindNextNeighbors)
```

File: day_10/task_1/pipe_maze.hpp [hunk 3]
```diff
@@ -4,4 +4,6 @@
 #include <vector>
 
 using PipeMaze = std::vector<std::string>;
+
```
Comment (Rosein) [inline day_10/task_1/pipe_maze.hpp]:
```suggestion
```

**Review by Rosein — CHANGES_REQUESTED** [review]
I have just minor suggestions

File: day_10/task_1/tests/pipe_maze_tests.cpp [hunk 4]
```diff
@@ -5,5 +5,14 @@ TEST(PipeMazeToolsShould, FindTheStartingPoint)
...
+{
+    const PipeMaze pipe_maze = {"-L|F7", "7S-7|", "L|7||", "-L-J|", "L|-JF"};
+
+    const auto foundNeighbors = find_next_neighbors(pipe_maze);
```
Comment (Rosein) [inline day_10/task_1/tests/pipe_maze_tests.cpp]: Shouldn't we be able to find the neighbors of specific pipe?
If we couldn't choose specific pipe, for which we would like to find neighbors, then I don't see the way, how we could use this function later.

```suggestion
    const MazePoint current_pipe(1,1);
    const auto foundNeighbors = find_next_neighbors(pipe_maze, current_pipe);
```

File: day_10/task_1/tests/pipe_maze_tests.cpp [hunk 5]
```diff
@@ -5,5 +5,16 @@ TEST(PipeMazeToolsShould, FindTheStartingPoint)
 {
     const PipeMaze pipe_maze = {"-L|F7", "7S-7|", "L|7||", "-L-J|", "L|-JF"};
 
-    ASSERT_EQ(find_starting_point(pipe_maze), (std::make_pair<int, int>(1, 1)));
+    ASSERT_EQ(find_starting_point(pipe_maze), MazePoint(1, 1));
+}
+
+TEST(PipeMazeToolsShould, FindNextNeighbors)
```
Comment (Rosein) [inline day_10/task_1/tests/pipe_maze_tests.cpp]:
```suggestion
TEST(PipeMazeToolsShould, GivenStartingPipe_WhenFindNextNeighbors_ExpectTwoMazePointsOfConnectedPipes)
```

File: day_10/task_1/tests/pipe_maze_tests.cpp [hunk 6]
```diff
@@ -5,5 +5,16 @@ TEST(PipeMazeToolsShould, FindTheStartingPoint)
...
+{
+    const PipeMaze pipe_maze = {"-L|F7", "7S-7|", "L|7||", "-L-J|", "L|-JF"};
+    const MazePoint current_pipe(1,1);
```
Comment (Rosein) [inline day_10/task_1/tests/pipe_maze_tests.cpp]:
```suggestion
    const MazePoint starting_pipe(find_starting_point(pipe_maze));
```

File: day_10/task_1/tests/pipe_maze_tests.cpp [hunk 7]
```diff
@@ -5,5 +5,16 @@ TEST(PipeMazeToolsShould, FindTheStartingPoint)
 {
     const PipeMaze pipe_maze = {"-L|F7", "7S-7|", "L|7||", "-L-J|", "L|-JF"};
 
-    ASSERT_EQ(find_starting_point(pipe_maze), (std::make_pair<int, int>(1, 1)));
+    ASSERT_EQ(find_starting_point(pipe_maze), MazePoint(1, 1));
+}
+
+TEST(PipeMazeToolsShould, FindNextNeighbors)
+{
+    const PipeMaze pipe_maze = {"-L|F7", "7S-7|", "L|7||", "-L-J|", "L|-JF"};
```
Comment (Rosein) [inline day_10/task_1/tests/pipe_maze_tests.cpp]: `pipe_maze` is used in the previous test also. We could avoid DRY violation by creating test fixture with `pipe_maze`. Could you apply such fixture, please?

File: day_10/task_1/tests/pipe_maze_tests.cpp [hunk 8]
```diff
@@ -5,5 +5,16 @@ TEST(PipeMazeToolsShould, FindTheStartingPoint)
 {
     const PipeMaze pipe_maze = {"-L|F7", "7S-7|", "L|7||", "-L-J|", "L|-JF"};
 
-    ASSERT_EQ(find_starting_point(pipe_maze), (std::make_pair<int, int>(1, 1)));
+    ASSERT_EQ(find_starting_point(pipe_maze), MazePoint(1, 1));
```
Comment (Rosein) [inline day_10/task_1/tests/pipe_maze_tests.cpp]: If you refactor this line, then I think for consistency you should also refactor the function find_the_starting_point and apply MazePoint, where it is applicable.

**Review by Rosein — CHANGES_REQUESTED** [review]
Few more suggestions :)

File: day_10/task_1/pipe_maze.cpp [hunk 9]
```diff
@@ -1,16 +1,16 @@
...
             if(pipe_maze[i][j] == 'S')
             {
-                return std::make_pair(i, j);
+                return MazePoint(i, j);
             }
         }
     }
-    return std::make_pair(-1, -1);
+    return MazePoint(-1, -1);
 }
```
Comment (Rosein) [inline day_10/task_1/pipe_maze.cpp]: Missing newline

File: .bazelrc [hunk 10]
```diff
@@ -1,3 +1,3 @@
 # Enable Bzlmod for every Bazel command
 # common --enable_bzlmod
-build --action_env=BAZEL_CXXOPTS="-std=c++20:-Werror:-DLOCAL_PATH=/home/rkumanek/Repos"
+build --action_env=BAZEL_CXXOPTS="-std=c++20:-Werror:-DLOCAL_PATH=/Users/dariakumanek/Desktop/workspace/projects"
```
Comment (Rosein) [inline .bazelrc]: Please revert this change

File: day_10/task_1/tests/pipe_maze_tests.cpp [hunk 11]
```diff
@@ -1,9 +1,23 @@
...
+TEST_F(PipeMazeTests, FindTheStartingPoint)
+{
+    ASSERT_EQ(find_starting_point(pipe_maze), MazePoint(1, 1));
+}
+
+TEST_F(PipeMazeTests, GivenStartingPipe_WhenFindNextNeighbors_ExpectTwoMazePointsOfConnectedPipes)
+{
+    const MazePoint starting_pipe(find_starting_point(pipe_maze));
+
+    const auto foundNeighbors = find_next_neighbors(pipe_maze, starting_pipe);
```
Comment (Rosein) [inline day_10/task_1/tests/pipe_maze_tests.cpp:19]: `starting_pipe`, `foundNeighbors` should be in same style case.

Could you allign to style one common style these two names, please?

**Review by Rosein — CHANGES_REQUESTED** [review]
Few minor suggestions :)

File: day_10/task_1/tests/pipe_maze_tests.cpp [hunk 12]
```diff
@@ -1,9 +1,23 @@
...
 #include "gtest/gtest.h"
 
-TEST(PipeMazeToolsShould, FindTheStartingPoint)
+
+struct PipeMazeTests : public ::testing::Test 
 {
     const PipeMaze pipe_maze = {"-L|F7", "7S-7|", "L|7||", "-L-J|", "L|-JF"};
+};
+
+TEST_F(PipeMazeTests, FindTheStartingPoint)
```
Comment (Rosein) [inline day_10/task_1/tests/pipe_maze_tests.cpp]:
```suggestion
TEST_F(PipeMazeTests, GivenMazeWithPipeLoop_WhenFindTheStartingPoint_ExpectCoordinatesFromFieldWhereS)
```

File: day_10/task_1/tests/pipe_maze_tests.cpp [hunk 13]
```diff
@@ -1,9 +1,23 @@
...
 #include "gtest/gtest.h"
 
-TEST(PipeMazeToolsShould, FindTheStartingPoint)
+
+struct PipeMazeTests : public ::testing::Test 
 {
     const PipeMaze pipe_maze = {"-L|F7", "7S-7|", "L|7||", "-L-J|", "L|-JF"};
+};
+
+TEST_F(PipeMazeTests, GivenStartingPipe_WhenFindNextNeighbors_ExpectTwoMazePointsOfConnectedPipes)
```
Comment (Rosein) [inline day_10/task_1/tests/pipe_maze_tests.cpp]:
```suggestion
TEST_F(PipeMazeTests, GivenMazeWithPipeLoop_WhenFindTheStartingPoint_ExpectCoordinatesFromFieldWhereS)
```

Comment (dariannax3) [reply day_10/task_1/tests/pipe_maze_tests.cpp]: done

File: day_10/task_1/tests/pipe_maze_tests.cpp (hunk 12 above)
Comment (dariannax3) [reply day_10/task_1/tests/pipe_maze_tests.cpp]: done

File: day_10/task_1/tests/pipe_maze_tests.cpp (hunk 11 above)
Comment (dariannax3) [reply day_10/task_1/tests/pipe_maze_tests.cpp:19]: done

File: day_10/task_1/tests/pipe_maze_tests.cpp (hunk 5 above)
Comment (dariannax3) [reply day_10/task_1/tests/pipe_maze_tests.cpp]: done

File: day_10/task_1/tests/pipe_maze_tests.cpp (hunk 7 above)
Comment (dariannax3) [reply day_10/task_1/tests/pipe_maze_tests.cpp]: done

File: day_10/task_1/pipe_maze.cpp (hunk 9 above)
Comment (dariannax3) [reply day_10/task_1/pipe_maze.cpp]: done

File: .bazelrc (hunk 10 above)
Comment (dariannax3) [reply .bazelrc]: done

File: day_10/task_1/pipe_maze.cpp (hunk 9 above)
Comment (Rosein) [reply day_10/task_1/pipe_maze.cpp]: Is it really done? It seems to still miss. Could you add it, please? :)

Comment (dariannax3) [reply day_10/task_1/pipe_maze.cpp]: i missed that, sorry

**Review by Rosein — APPROVED** [review]
Good job! :)
